backend/src/index.ts

```diff
@@ -1,16 +1,64 @@
 import express, { Response } from "express";
 import session from "express-session";
-import { TypedRequest, LoginBody } from "./requestTypes";
+import { TypedRequest, LoginBody, ValidateBody } from "./requestTypes";
+import { SessionStorage, User } from "./interfaces";
 import bcrypt from "bcrypt";
-import { collection, addDoc, getDocs, where, query } from "firebase/firestore";
+import {
+  collection,
+  addDoc,
+  getDocs,
+  where,
+  query,
+  deleteDoc,
+} from "firebase/firestore";
 import { db } from "./firebase";
 import cors from "cors";
 import crypto from "crypto";
 
-const EXPRESS_PORT = 3000;
+const session_auth = async (sessionId: string): Promise<boolean> => {
+  const sessions = collection(db, "sessions");
+  const sessionData = query(sessions, where("sessionId", "==", sessionId));
+  const session = await getDocs(sessionData);
+
+  if (session.empty) {
+    return false;
+  }
+
+  // if current time is greater than expiration date, return false and
+  // removes it from the database
+  if (new Date() > session.docs[0].data().expirationDate.toDate()) {
+    console.log(new Date(), session.docs[0].data().expirationDate);
+    await session_remove(sessionId);
+    return false;
+  }
+  return true;
+};
+
+// Clears all expired sessions in the db
+// Idea, periodically call this to stay efficient on document reads?
+const clean_sessions = async (): Promise<void> => {
```

Comment:
this function doesnt seem to really take advantage of returning a promise

Reply:
so a func can be async without containing a promise?

Reply:
sorry what i mean is you should be able to get away with not specifying the return type here